forc-plugins/forc-doc/src/render/index.rs

```diff
@@ -0,0 +1,243 @@
+                        //                 input(
+                        //                     class="search-input",
+                        //                     name="search",
+                        //                     autocomplete="off",
+                        //                     spellcheck="false",
+                        //                     // TODO: Add functionality.
+                        //                     placeholder="Searchbar unimplemented, see issue #3480...",
+                        //                     type="search"
+                        //                 );
+                        //                 div(id="help-button", title="help", tabindex="-1") {
+                        //                     button(type="button") { : "?" }
+                        //                 }
+                        //             }
+                        //         }
+                        //     }
+                        // }
+                        section(id="main-content", class="content") {
+                            h1(class="fqn") {
+                                span(class="in-band") { : "List of all items" }
+                            }
+                            : doc_links;
+                        }
+                    }
+                }
+                script(src="assets/highlight.js");
+                script {
+                    : "hljs.highlightAll();";
+                }
+            }
+        })
+    }
+}
+
+/// The index for each module in a Sway project.
+pub(crate) struct ModuleIndex {
+    /// used only for the root module
+    version_opt: Option<String>,
+    module_info: ModuleInfo,
+    module_docs: DocLinks,
+}
+impl ModuleIndex {
+    pub(crate) fn new(
+        version_opt: Option<String>,
+        module_info: ModuleInfo,
+        module_docs: DocLinks,
+    ) -> Self {
+        ModuleIndex {
+            version_opt,
+            module_info,
+            module_docs,
+        }
+    }
+}
+impl SidebarNav for ModuleIndex {
+    fn sidebar(&self) -> Sidebar {
+        let style = match self.module_info.is_root_module() {
+            true => self.module_docs.style.clone(),
+            false => DocStyle::ModuleIndex,
+        };
+        Sidebar::new(
+            self.version_opt.clone(),
+            style,
+            self.module_info.clone(),
+            self.module_docs.clone(),
+        )
+    }
+}
+impl Renderable for ModuleIndex {
+    fn render(self, render_plan: RenderPlan) -> Result<Box<dyn RenderBox>> {
+        let doc_links = self.module_docs.clone().render(render_plan.clone())?;
+        let sidebar = self.sidebar().render(render_plan)?;
+        let title_prefix = match self.module_docs.style {
+            DocStyle::ProjectIndex(ref program_type) => format!("{program_type} "),
+            DocStyle::ModuleIndex => "Module ".to_string(),
+            _ => unreachable!("Module Index can only be either a project or module at this time."),
```

Comment:
Unreachable over panic?

Reply:
Again, used in instances where I used a match statement and all other types to match on will never be reached